modules/ROOT/pages/dataweave-functions.adoc

```diff
@@ -7,9 +7,135 @@ You can define your own DataWeave functions using the `fun` declaration in the h
 //LINK TO DW 1.0 LANDING PAGE IN MULE 3.9 DOCS:
 include::partial$dataweave1-links.adoc[tag=dataweave1LandingPage]
 
+== DataWeave Function Definition Syntax
 
-In this simple example, a DataWeave function accepts a single String argument that outputs `"HELLO"`:
+To define a function in DataWeave use the following syntax:
+----
+fun myFunction(param1, param2, ...) = <code to execute>
+----
+
+* The `fun` keyword starts the definition of a function.
+* `myFunction` is the name you define for the function.
++
+Function names must be valid identifiers. For additional details about valid identifiers, see xref:dataweave-language-introduction.adoc#valid-identifiers[Rules for Declaring Valid Identifiers].
+* `(param1, param2, ... , paramn)` represents the parameters that your function accepts.
++
+You can define from zero to any number of parameters, separated by commas (`,`) and enclosed in parenthesis.
+* The `=` sign marks the beginning of the code block to execute when the function is called.
+* `<code to execute>` represents the actual code that you define for your function.
+
```

Comment:
Maybe we could add something about the syntax using the type system constraints. Not sure if as another section or in this one. https://docs.mulesoft.com/mule-runtime/4.3/dataweave-type-system

Something like this?
```
To take advantage of the type-checking that the DataWeave type system executes, you could provide constraint expression functions. 
These constraints are used in the type-checking phase when DataWeave ensures that the arguments for a function call respect its constraints.

To define constraints for a function use the following syntax:
----
fun myFunction(param1: Type, param2: Type): ResultType = <code to execute>
----

For example:
----
%dw 2.0
output application/json
fun toUser(id: Number, userName: String): String = "you called the function toUser!"
----

The constraints to call the function toUser are that the first argument has to be of type Number, the second one of type String (to be coercible to those types). Another constraint is that the result of the function has to be of type String. This constraint isn’t used in the call for toUser; it’s applied to its definition (`<code to execute>`) to validate that the function body generates the proper type.

You can find more info about the DataWeave Type System in https://docs.mulesoft.com/mule-runtime/4.3/dataweave-type-system.
```